src/client/datascience/ipywidgets/ipyWidgetMessageDispatcher.ts

```diff
@@ -178,12 +178,12 @@ export class IPyWidgetMessageDispatcher implements IIPyWidgetMessageDispatcher {
         this._postMessageEmitter.fire({ message, payload });
     }
     private subscribeToKernelSocket(kernel: IKernel) {
-        if (this.subscribedToKernelSocket || !kernel.notebook) {
+        if (this.subscribedToKernelSocket || !kernel.session) {
             return;
         }
         this.subscribedToKernelSocket = true;
         // Listen to changes to kernel socket (e.g. restarts or changes to kernel).
-        kernel.notebook.session.kernelSocket.subscribe((info) => {
+        kernel.session.kernelSocket.subscribe((info) => {
```

Comment:
Using session directly, INotebook was just a wrapper around IJUpyterSession, and we can use IJupyterSession directory. Also the plan is to get IJupyterSession line up with jupyters `Krenel.IKernelCOnnection` (& it is very close)